verification/src/http_server/handlers/verification/sourcify/metadata.rs

```diff
@@ -0,0 +1,187 @@
+use std::collections::BTreeMap;
+
+use ethers_solc::EvmVersion;
+use serde::Deserialize;
+
+use crate::VerificationResult;
+
+use super::types::Files;
+
+const METADATA_FILE_NAME: &str = "metadata.json";
+
+#[derive(Debug, PartialEq, Deserialize)]
+
+pub struct Metadata {
+    pub settings: MetadataSettings,
+    pub compiler: Compiler,
+    pub output: Output,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetadataSettings {
+    pub compilation_target: BTreeMap<String, String>,
+    pub optimizer: Optimizer,
+    pub libraries: BTreeMap<String, String>,
+    #[serde(rename = "camelCase")]
+    pub evm_version: Option<EvmVersion>,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct Optimizer {
+    pub enabled: Option<bool>,
```

Comment:
I'm not sure why you're declaring this, but you might wanna look into ether_solc::artifacts (but I think you've already looked)